tests/phpunit/tests/test-admin-filters.php

```diff
@@ -17,27 +17,39 @@ function setUp() {
 	}
 
 	function test_sanitize_title() {
-		self::$polylang->filters->curlang = self::$polylang->model->get_language( 'en' );
+		$pll_admin = new PLL_Admin( self::$polylang->links_model );
+		$pll_admin->curlang = self::$polylang->model->get_language( 'en' );
+		$pll_admin->add_filters();
 		$this->assertEquals( 'fullmenge', sanitize_title( 'Füllmenge' ) );
 
-		self::$polylang->filters->curlang = self::$polylang->model->get_language( 'de' );
+		$pll_admin = new PLL_Admin( self::$polylang->links_model );
+		$pll_admin->curlang = self::$polylang->model->get_language( 'de' );
+		$pll_admin->add_filters();
 		$this->assertEquals( 'fuellmenge', sanitize_title( 'Füllmenge' ) );
 
 		unset( self::$polylang->filters->curlang );
 
 		// Bug fixed in 2.4.1
 		$_POST['post_lang_choice'] = 'en';
+		$pll_admin = new PLL_Admin( self::$polylang->links_model );
+		$pll_admin->add_filters();
 		$this->assertEquals( 'fullmenge', sanitize_title( 'Füllmenge' ) );
 
 		$_POST['post_lang_choice'] = 'de';
+		$pll_admin = new PLL_Admin( self::$polylang->links_model );
+		$pll_admin->add_filters();
 		$this->assertEquals( 'fuellmenge', sanitize_title( 'Füllmenge' ) );
 	}
 
 	function test_sanitize_user() {
-		self::$polylang->filters->curlang = self::$polylang->model->get_language( 'en' );
+		$pll_admin = new PLL_Admin( self::$polylang->links_model );
+		$pll_admin->curlang = self::$polylang->model->get_language( 'en' );
+		$pll_admin->add_filters();
 		$this->assertEquals( 'angstrom', sanitize_user( 'ångström' ) );
 
-		self::$polylang->filters->curlang = self::$polylang->model->get_language( 'de' );
+		$pll_admin = new PLL_Admin( self::$polylang->links_model );
+		$pll_admin->curlang = self::$polylang->model->get_language( 'de' );
+		$pll_admin->add_filters();
```

Comment:
After applying the @Chouby's suggestion https://github.com/polylang/polylang/pull/552#discussion_r492638303 you're right: test_sanitize_title() breaks at the third assertion because the sanitization filter is still available.

I think it's interesting to keep all the case - and so assertions - in the same method. So I put a remove_filter() call before each case.

Reply:
After talking about during the last DM. I splitted these unit tests to separate each case.